wrapper/src/main/java/software/amazon/jdbc/Driver.java

```diff
@@ -96,10 +96,21 @@ public Connection connect(final String url, final Properties info) throws SQLExc
     LOGGER.finest("Opening connection to " + url);
 
     final String driverUrl = url.replaceFirst(PROTOCOL_PREFIX, "jdbc:");
-    final java.sql.Driver driver = DriverManager.getDriver(driverUrl);
+
+    final String message = Messages.get(
+        "Driver.missingDriver",
+        new Object[] {driverUrl, Collections.list(DriverManager.getDrivers())});
+
+    java.sql.Driver driver;
+    try {
+      driver = DriverManager.getDriver(driverUrl);
+    } catch (SQLException e) {
+      LOGGER.warning(() -> message);
```

Comment:
isn't it better to create the message here, once something was caught?
a bit odd to create the message before the try/catch